File: Python/prints.py
# ======= أساسي =======
print("Hello, World!")  # طباعة نص
print(5)                # طباعة رقم
print(3.14)             # طباعة رقم عشري
print(True)             # طباعة قيمة منطقية

# ======= طباعة عدة قيم =======
name = "Osama"
age = 20
print("Name:", name, "Age:", age)  # طباعة أكثر من قيمة

# ======= دمج النصوص والأرقام =======
# الطريقة 1: باستخدام الفاصلة (`,` تفصل بين العناصر)
print("Age is", age)

# الطريقة 2: باستخدام الـ f-string (أكثر شيوعاً في بايثون 3.6+)
print(f"My name is {name} and I am {age} years old.")

# الطريقة 3: باستخدام الـ format()
print("My name is {} and I am {} years old.".format(name, age))

# ======= الطباعة بدون سطر جديد =======
print("Hello", end=" ")  # end="" يمنع الانتقال للسطر الجديد
print("World")           # النتيجة: Hello World

# ======= الطباعة مع فاصل محدد =======
print("A", "B", "C", sep="-")  # النتيجة: A-B-C

# ======= طباعة القيم من العمليات =======
x = 5
y = 3
print("Sum:", x + y)      # Sum: 8
print("Product:", x * y)  # Product: 15

# ======= الطباعة من القوائم =======
fruits = ["apple", "banana", "orange"]
print("Fruits list:", fruits)

# ======= الطباعة من القواميس =======
student = {"name": "Osama", "age": 20}
print("Student info:", student)

File: dart/prints.dart
// =====================
// Prints in Dart
// =====================

// ignore_for_file: avoid_print

void main() {
  // ======== طباعة نصوص ========
  print('Hello, Dart!');

  // ======== طباعة أرقام ========
  print(2025);
  print(3.14);

  // ======== طباعة متغيرات ========
  String name = 'Osama';
  int age = 22;
  print(name);
  print(age);

  // ======== دمج النصوص مع المتغيرات ========
  print('My name is $name and I am $age years old.');

  // ======== استخدام ${} للعمليات ========
  print('Next year I will be ${age + 1} years old.');

  // ======== طباعة متعددة الأسطر ========
  print('''This is line 1
This is line 2
This is line 3''');

  // ======== استخدام الرموز الخاصة ========
  print('Column1\tColumn2\tColumn3'); // tab
  print('Line A\nLine B\nLine C'); // new line

  // ======== الطباعة الشرطية ========
  bool isAdmin = true;
  // ignore: dead_code
  print(isAdmin ? 'Welcome Admin' : 'Welcome User');

  // ======== طباعة القوائم والقواميس ========
  var list = [1, 2, 3];
  var map = {'a': 1, 'b': 2};
  print(list);
  print(map);

  // ======== طباعة الوقت الحالي ========
  print('Now: ${DateTime.now()}');
}
